src/main/java/au/org/ala/biocache/util/thread/DownloadControlThread.java

```diff
@@ -0,0 +1,165 @@
+    
+    private final DownloadServiceExecutor downloadServiceExecutor;
+    private final AtomicBoolean shutdownFlag = new AtomicBoolean(false);
+    private final Queue<DownloadDetailsDTO> currentDownloads;
+    private final DownloadCreator downloadCreator;
+    private final PersistentQueueDAO persistentQueueDAO;
+    
+    public DownloadControlThread(Integer maxRecords, DownloadType downloadType, int concurrencyLevel, Long pollDelayMs, Long executionDelayMs, Integer threadPriority, Queue<DownloadDetailsDTO> currentDownloads, DownloadCreator downloadCreator, PersistentQueueDAO persistentQueueDAO) {
+        this.maxRecords = maxRecords;
+        this.downloadType = downloadType;
+        this.concurrencyLevel = concurrencyLevel > 0 ? concurrencyLevel : 1;
+        this.pollDelay = pollDelayMs != null && pollDelayMs >= 0L ? pollDelayMs : 10L;
+        this.executionDelay = executionDelayMs != null && executionDelayMs >= 0L ? executionDelayMs : 0L;
+        this.threadPriority = threadPriority != null && threadPriority >= Thread.MIN_PRIORITY && threadPriority <= Thread.MAX_PRIORITY ? threadPriority : Thread.NORM_PRIORITY;
+        this.currentDownloads = currentDownloads;
+        this.downloadCreator = downloadCreator;
+        this.persistentQueueDAO = persistentQueueDAO;
+        // Create a dedicated ExecutorService for this thread
+        this.downloadServiceExecutor = createExecutor();
+    }
+
+    protected DownloadServiceExecutor createExecutor() {
+        return new DownloadServiceExecutor(this.maxRecords, this.downloadType, 
+                                           this.concurrencyLevel, this.executionDelay, 
+                                           this.threadPriority, this.downloadCreator);
+    }
+
+    @Override
+    public void run() {
+        DownloadDetailsDTO currentDownload = null;
+        try {
+            // Runs until the thread is interrupted, then shuts down all of the downloads under its control before returning
+            while (true) {
+                if(shutdownFlag.get() || Thread.currentThread().isInterrupted()) {
+                    break;
+                }
+                if(!downloadServiceExecutor.reserveCapacity(pollDelay, TimeUnit.MILLISECONDS)) {
+                    // If we couldn't reserve capacity within this period, go back around and try again after checking the interrupt status
+                    continue;
+                }
+                if(shutdownFlag.get() || Thread.currentThread().isInterrupted()) {
+                    break;
+                }
+                currentDownload = persistentQueueDAO.getNextDownload(maxRecords, downloadType);
+                if (currentDownload != null) {
+                    downloadServiceExecutor.submitDownload(currentDownload);
+                } else {
+                    // We need to return the capacity we reserved because we don't need to use it
+                    downloadServiceExecutor.returnCapacity();
```

Comment:
There is no callable created if it reaches the else branch here, but we already reserved capacity on the semaphore before checking the queue and finding no matches on the download queue. 

It would need to get into submitDownload (the if branch above) for the callable to be created, which will indeed release the capacity after it has sent the email after completing (successfully or not).